tutorials/tools/Text_Normalization_Tutorial.ipynb

```diff
@@ -0,0 +1,472 @@
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yyUE_t4vw2et"
+      },
+      "source": [
+        "The `DATA_DIR` should now contain the text file"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "VXrTzTyIpzE8"
+      },
+      "source": [
+        "!ls -l $DATA_DIR"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "FWqlbSryw_WL"
+      },
+      "source": [
+        "print the first 10 lines of the file :"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "1vC2DHawIGt8"
+      },
+      "source": [
+        "! head -n 10 $DATA_DIR/$text_file"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RMT5lkPYzZHK"
+      },
+      "source": [
+        "## Data for Evaluation\n",
+        "\n",
+        "The data for Evaluation needs to be segmented and labeled by semiotic class, following the format of [Google Text normalization dataset](https://www.kaggle.com/richardwilliamsproat/text-normalization-for-english-russian-and-polish).\n",
+        "That is, every line of the file needs to have the format <semiotic class>\\t<unnormalized text>\\t<`self` if trivial class or normalized text>.\n",
+        "The semiotic classes are the same as in the Google Text normalization dataset:\n",
+        "PLAIN, PUNCT, DATE, CARDINAL, LETTERS, VERBATIM, MEASURE, DECIMAL, ORDINAL, DIGIT, MONEY, TELEPHONE, ELECTRONIC, FRACTION, TIME, ADDRESS.\n",
+        "\n",
+        "We additionally added the class `WHITELIST` for all whitelisted tokens whose verbalizations directly looked up from `$TOOLS_DATA_DIR/whitelist.tsv`. To extend the list simply add further key value pairs to the file.\n",
+        "\n",
+        "\n",
+        "We will create a simple example file extracted from the Google Text normalization dataset to show how evaluation works:\n",
+        "\n",
+        "\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "u4zjeVVv-UXR"
+      },
+      "source": [
+        "eval_input_data =  \"\"\"LETTERS\\tA & E\\ta and e\r\n",
+        "PUNCT\\t.\\tsil\r\n",
+        "PLAIN\\tRetrieved\\t<self>\r\n",
+        "DATE\\t2006-08-05\\tthe fifth of august two thousand six\r\n",
+        "PUNCT\\t.\\tsil\r\n",
+        "<eos>\\t<eos>\r\n",
+        "PLAIN\\tDownloaded\\t<self>\r\n",
+        "PLAIN\\ton\\t<self>\r\n",
+        "DATE\\t7 August 2007\\tthe seventh of august two thousand seven\r\n",
+        "PUNCT\\t.\\tsil\"\"\"\r\n",
+        "eval_text_file_path = f\"{DATA_DIR}/00001-of-00100\"\r\n",
+        "with open(eval_text_file_path, 'w') as fp:\r\n",
+        "  fp.write(eval_input_data)\r\n",
+        "! cat $eval_text_file_path\r\n"
+      ],
```

Comment:
It's a bit confusing that we're creating this snippet here. Maybe rearrange the sections: Download the data to predictions -> Prediction -> Download the data for evaluation -> Evaluation ?

Reply:
done